sql-cli/sql_cli/utils/airflow.py

```diff
@@ -24,61 +24,56 @@ def airflow_version() -> Version:
     return Version(airflow.__version__)
 
 
-def retrieve_airflow_database_conn_from_config(airflow_home: Path) -> str:
+def initialise(airflow_home: Path, airflow_dags_folder: Path) -> None:
     """
-    Retrieve the path to the Airflow metadata database connection.
+    Create an Airflow database and configure airflow via environment variables.
 
-    :params airfow_home: Path to where Airflow was initialized ($AIRFLOW_HOME).
-    :returns: Airflow metadata database connection URI
+    :params airflow_home: The airflow home to set
+    :params airflow_dags_folder: The airflow dags folder to set
     """
-    os.environ["AIRFLOW_HOME"] = str(airflow_home.resolve())
-    filename = airflow_home / "airflow.cfg"
-    parser = ConfigParser()
-    parser.read(filename)
-    confdict = {section: dict(parser.items(section)) for section in parser.sections()}
-
-    if airflow_version() >= Version("2.3"):
-        sql_alchemy_conn = confdict["database"]["sql_alchemy_conn"]
+    # Set airflow environment variables prior to database initialization
+    os.environ["AIRFLOW_HOME"] = airflow_home.as_posix()
+    os.environ["AIRFLOW__CORE__DAGS_FOLDER"] = airflow_dags_folder.as_posix()
+    if version() >= Version("2.3"):
+        os.environ.pop("AIRFLOW__DATABASE__SQL_ALCHEMY_CONN", None)
     else:
-        sql_alchemy_conn = confdict["core"]["sql_alchemy_conn"]
+        os.environ.pop("AIRFLOW__CORE__SQL_ALCHEMY_CONN", None)
 
-    return sql_alchemy_conn
+    # Reload airflow configuration after setting environment variables
+    from airflow import configuration
 
+    importlib.reload(configuration)
 
-def disable_examples(airflow_home: Path) -> None:
-    """
-    Disable Airflow examples in the configuration file available at the given airflow_home directory.
-
-    :params airflow_home: Path to where Airflow was initialised ($AIRFLOW_HOME)
-    """
-    filename = airflow_home / "airflow.cfg"
-    parser = ConfigParser()
-    parser.read(filename)
-    parser["core"]["load_examples"] = "False"
-    parser["core"]["logging_level"] = "WARN"
-    with open(filename, "w") as out_fp:
-        parser.write(out_fp)
+    # Initialise the airflow database & hide all logs
+    os.system("airflow db init > /dev/null 2>&1")  # skipcq: BAN-B605,BAN-B607
```

Comment:
Should we improve this in the verbose/debug ticket - so we can have more visibility if we use the flag?

Reply:
Yes, great idea! I will add this to the ticket.